File: NegocioCSharp/PerfilMenu.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
namespace LiquidacionSueldos
{
    namespace Negocio
    {
        public partial class PerfilMenu
        {
            #region Propiedades
            private Datos.Gestor ocdGestor = new Datos.Gestor();
            private DataTable Fila;

            private DataTable Tabla = new DataTable();
            private int _pmeId;
            public int pmeId
            {
                get { return _pmeId; }
                set { _pmeId = value; }
            }

            private bool _pmeActivo;
            public bool pmeActivo
            {
                get { return _pmeActivo; }
                set { _pmeActivo = value; }
            }

            private System.DateTime _pmeFechaHoraCreacion;
            public System.DateTime pmeFechaHoraCreacion
            {
                get { return _pmeFechaHoraCreacion; }
                set { _pmeFechaHoraCreacion = value; }
            }

            private System.DateTime _pmeFechaHoraUltimaModificacion;
            public System.DateTime pmeFechaHoraUltimaModificacion
            {
                get { return _pmeFechaHoraUltimaModificacion; }
                set { _pmeFechaHoraUltimaModificacion = value; }
            }

            private int _perId;
            public int perId
            {
                get { return _perId; }
                set { _perId = value; }
            }

            private int _menId;
            public int menId
            {
                get { return _menId; }
                set { _menId = value; }
            }

            private int _usuIdCreacion;
            public int usuIdCreacion
            {
                get { return _usuIdCreacion; }
                set { _usuIdCreacion = value; }
            }

            private int _usuIdUltimaModificacion;
            public int usuIdUltimaModificacion
            {
                get { return _usuIdUltimaModificacion; }
                set { _usuIdUltimaModificacion = value; }
            }
            #endregion 

            public PerfilMenu()
            {
                try
                {
                    this.pmeId = 0;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public PerfilMenu(int pmeId)
            {
                try
                {
                    Fila = new DataTable();
                    Fila = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerUno]", new object[,] { {
                        "@pmeId",
                        pmeId
                    } });

                    if (Fila.Rows.Count > 0)
                    {
                        if (Fila.Rows[0]["pmeId"].ToString().Trim().Length > 0)
                        {
                            this.pmeId = Convert.ToInt32(Fila.Rows[0]["pmeId"]);
                        }
                        else
                        {
                            this.pmeId = 0;
                        }

                        if (Fila.Rows[0]["pmeFechaHoraCreacion"].ToString().Trim().Length > 0)
                        {
                            this.pmeFechaHoraCreacion = Convert.ToDateTime(Fila.Rows[0]["pmeFechaHoraCreacion"]);
                        }
                        else
                        {
                            this.pmeFechaHoraCreacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["pmeFechaHoraUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.pmeFechaHoraUltimaModificacion = Convert.ToDateTime(Fila.Rows[0]["pmeFechaHoraUltimaModificacion"]);
                        }
                        else
                        {
                            this.pmeFechaHoraUltimaModificacion = System.DateTime.Now;
                        }

                        if (Fila.Rows[0]["pmeActivo"].ToString().Trim().Length > 0)
                        {
                            this.pmeActivo = (Convert.ToInt32(Fila.Rows[0]["pmeActivo"]) == 1 ? true : false);
                        }
                        else
                        {
                            this.pmeActivo = false;
                        }

                        if (Fila.Rows[0]["perId"].ToString().Trim().Length > 0)
                        {
                            this.perId = Convert.ToInt32(Fila.Rows[0]["perId"]);
                        }
                        else
                        {
                            this.perId = 0;
                        }

                        if (Fila.Rows[0]["menId"].ToString().Trim().Length > 0)
                        {
                            this.menId = Convert.ToInt32(Fila.Rows[0]["menId"]);
                        }
                        else
                        {
                            this.menId = 0;
                        }

                        if (Fila.Rows[0]["usuIdCreacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdCreacion = Convert.ToInt32(Fila.Rows[0]["usuIdCreacion"]);
                        }
                        else
                        {
                            this.usuIdCreacion = 0;
                        }

                        if (Fila.Rows[0]["usuIdUltimaModificacion"].ToString().Trim().Length > 0)
                        {
                            this.usuIdUltimaModificacion = Convert.ToInt32(Fila.Rows[0]["usuIdUltimaModificacion"]);
                        }
                        else
                        {
                            this.usuIdUltimaModificacion = 0;
                        }

                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public PerfilMenu(int pmeId, bool pmeActivo, System.DateTime pmeFechaHoraCreacion, System.DateTime pmeFechaHoraUltimaModificacion, int IperId, int ImenId, int IusuIdCreacion, int IusuIdUltimaModificacion)
            {
                try
                {
                    this.pmeId = pmeId;
                    this.pmeActivo = pmeActivo;
                    this.pmeFechaHoraCreacion = pmeFechaHoraCreacion;
                    this.pmeFechaHoraUltimaModificacion = pmeFechaHoraUltimaModificacion;
                    this.perId = perId;
                    this.menId = menId;
                    this.usuIdCreacion = usuIdCreacion;
                    this.usuIdUltimaModificacion = usuIdUltimaModificacion;
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }


            public DataTable ObtenerTodo()
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerTodo]", new object[,]{});
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerUno(int pmeId)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerUno]", new object[,] { {
                        "@pmeId",
                        pmeId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public DataTable ObtenerxperIdCustom(int perId)
            {
                ocdGestor = new Datos.Gestor();
                Tabla = new DataTable();

                try
                {
                    Tabla = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerxperIdCustom]", new object[,] { {
                        "@perId",
                        perId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }

                return Tabla;
            }

            public void Actualizar(int pmeId, int perId, int menId, bool pmeActivo, int usuIdCreacion, int usuIdUltimaModificacion, System.DateTime pmeFechaHoraCreacion, System.DateTime pmeFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.Actualizar]", new object[,] {
                        {
                            "@pmeId",
                            pmeId
                        },
                        {
                            "@perId",
                            perId
                        },
                        {
                            "@menId",
                            menId
                        },
                        {
                            "@pmeActivo",
                            pmeActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@pmeFechaHoraCreacion",
                            pmeFechaHoraCreacion
                        },
                        {
                            "@pmeFechaHoraUltimaModificacion",
                            pmeFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar(int pmeId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.Eliminar]", new object[,] { {
                        "@pmeId",
                        pmeId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void EliminarPadresSinHijos(int perId)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.EliminarPadresSinHijos]", new object[,] { {
                        "@perId",
                        perId
                    } });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Insertar(int perId, int menId, bool pmeActivo, int usuIdCreacion, int usuIdUltimaModificacion, System.DateTime pmeFechaHoraCreacion, System.DateTime pmeFechaHoraUltimaModificacion)
            {
                try
                {
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.Insertar]", new object[,] {
                        {
                            "@perId",
                            perId
                        },
                        {
                            "@menId",
                            menId
                        },
                        {
                            "@pmeActivo",
                            pmeActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@pmeFechaHoraCreacion",
                            pmeFechaHoraCreacion
                        },
                        {
                            "@pmeFechaHoraUltimaModificacion",
                            pmeFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Actualizar()
            {
                try
                {
                    if (this.pmeId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[PerfilMenu.Actualizar]", new object[,] {
                            {
                                "@pmeId",
                                pmeId
                            },
                            {
                                "@perId",
                                perId
                            },
                            {
                                "@menId",
                                menId
                            },
                            {
                                "@pmeActivo",
                                pmeActivo
                            },
                            {
                                "@usuIdCreacion",
                                usuIdCreacion
                            },
                            {
                                "@usuIdUltimaModificacion",
                                usuIdUltimaModificacion
                            },
                            {
                                "@pmeFechaHoraCreacion",
                                pmeFechaHoraCreacion
                            },
                            {
                                "@pmeFechaHoraUltimaModificacion",
                                pmeFechaHoraUltimaModificacion
                            }
                        });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public void Eliminar()
            {
                try
                {
                    if (this.pmeId != 0)
                    {
                        ocdGestor.EjecutarNonQuery("[PerfilMenu.Eliminar]", new object[,] { {
                            "@pmeId",
                            pmeId
                        } });
                    }
                }
                catch (Exception ex)
                {
                    throw ex;
                }
            }

            public int Insertar()
            {
                int IdMax = 0;
                try
                {
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[PerfilMenu.Insertar]", new object[,] {
                        {
                            "@perId",
                            perId
                        },
                        {
                            "@menId",
                            menId
                        },
                        {
                            "@pmeActivo",
                            pmeActivo
                        },
                        {
                            "@usuIdCreacion",
                            usuIdCreacion
                        },
                        {
                            "@usuIdUltimaModificacion",
                            usuIdUltimaModificacion
                        },
                        {
                            "@pmeFechaHoraCreacion",
                            pmeFechaHoraCreacion
                        },
                        {
                            "@pmeFechaHoraUltimaModificacion",
                            pmeFechaHoraUltimaModificacion
                        }
                    });
                }
                catch (Exception ex)
                {
                    throw ex;
                }
                return IdMax;
            }

        }
    }
}

File: Negocio/PerfilMenu.vb
Namespace LiquidacionSueldos
    Namespace Negocio
        Partial Public Class PerfilMenu
        
			Private ocdGestor As New LiquidacionSueldos.Datos.Gestor
            Private Fila As DataTable
            Private Tabla As New DataTable

			Private _pmeId As Integer
            Public Property pmeId() As Integer
                Get
					Return _pmeId
                End Get
                Set(ByVal value As Integer)
                    _pmeId = value
                End Set
            End Property

			Private _pmeActivo As Boolean
            Public Property pmeActivo() As Boolean
                Get
					Return _pmeActivo
                End Get
                Set(ByVal value As Boolean)
                    _pmeActivo = value
                End Set
            End Property

			Private _pmeFechaHoraCreacion As Date
            Public Property pmeFechaHoraCreacion() As Date
                Get
					Return _pmeFechaHoraCreacion
                End Get
                Set(ByVal value As Date)
                    _pmeFechaHoraCreacion = value
                End Set
            End Property

			Private _pmeFechaHoraUltimaModificacion As Date
            Public Property pmeFechaHoraUltimaModificacion() As Date
                Get
					Return _pmeFechaHoraUltimaModificacion
                End Get
                Set(ByVal value As Date)
                    _pmeFechaHoraUltimaModificacion = value
                End Set
            End Property

			Private _perId As Integer
            Public Property perId() As Integer
                Get
					Return _perId
                End Get
                Set(ByVal value As Integer)
                    _perId = value
                End Set
            End Property

			Private _menId As Integer
            Public Property menId() As Integer
                Get
					Return _menId
                End Get
                Set(ByVal value As Integer)
                    _menId = value
                End Set
            End Property

			Private _usuIdCreacion As Integer
            Public Property usuIdCreacion() As Integer
                Get
					Return _usuIdCreacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdCreacion = value
                End Set
            End Property

			Private _usuIdUltimaModificacion As Integer
            Public Property usuIdUltimaModificacion() As Integer
                Get
					Return _usuIdUltimaModificacion
                End Get
                Set(ByVal value As Integer)
                    _usuIdUltimaModificacion = value
                End Set
            End Property

			Public Sub New()
                Try                
                    Me.pmeId = 0
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub            

			Public Sub New(ByVal pmeId As Integer)
				Try
                    Fila = New DataTable
                    Fila = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerUno]", New Object(,) {{"@pmeId", pmeId}})

					If Fila.Rows.Count > 0 Then
						If Fila(0)("pmeId").ToString().Trim().Length > 0 Then
                            Me.pmeId = Convert.ToInt32(Fila(0)("pmeId"))
                        Else
                            Me.pmeId = 0
                        End If
                        
						If Fila(0)("pmeFechaHoraCreacion").ToString().Trim().Length > 0 Then
                            Me.pmeFechaHoraCreacion = Convert.ToDateTime(Fila(0)("pmeFechaHoraCreacion"))
                        Else
                            Me.pmeFechaHoraCreacion = Date.Now
                        End If
                        
						If Fila(0)("pmeFechaHoraUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.pmeFechaHoraUltimaModificacion = Convert.ToDateTime(Fila(0)("pmeFechaHoraUltimaModificacion"))
                        Else
                            Me.pmeFechaHoraUltimaModificacion = Date.Now
                        End If
                        
						If Fila(0)("pmeActivo").ToString().Trim().Length > 0 Then
                            Me.pmeActivo = IIf(Convert.ToInt32(Fila(0)("pmeActivo")) = 1, True, False)
                        Else
							Me.pmeActivo = False
                        End If
                        
						If Fila(0)("perId").ToString().Trim().Length > 0 Then
                            Me.perId = Convert.ToInt32(Fila(0)("perId"))
                        Else
							Me.perId = 0
                        End If
                        
						If Fila(0)("menId").ToString().Trim().Length > 0 Then
                            Me.menId = Convert.ToInt32(Fila(0)("menId"))
                        Else
							Me.menId = 0
                        End If
                        
						If Fila(0)("usuIdCreacion").ToString().Trim().Length > 0 Then
                            Me.usuIdCreacion = Convert.ToInt32(Fila(0)("usuIdCreacion"))
                        Else
							Me.usuIdCreacion = 0
                        End If
                        
						If Fila(0)("usuIdUltimaModificacion").ToString().Trim().Length > 0 Then
                            Me.usuIdUltimaModificacion = Convert.ToInt32(Fila(0)("usuIdUltimaModificacion"))
                        Else
							Me.usuIdUltimaModificacion = 0
                        End If
                        
					End If
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub New(ByVal pmeId As Integer, ByVal pmeActivo As Boolean, ByVal pmeFechaHoraCreacion As Date, ByVal pmeFechaHoraUltimaModificacion As Date, ByVal IperId As Integer, ByVal ImenId As Integer, ByVal IusuIdCreacion As Integer, ByVal IusuIdUltimaModificacion As Integer)
				Try
					me.pmeId = pmeId
					me.pmeActivo = pmeActivo
					me.pmeFechaHoraCreacion = pmeFechaHoraCreacion
					me.pmeFechaHoraUltimaModificacion = pmeFechaHoraUltimaModificacion
					me.perId = perId
					me.menId = menId
					me.usuIdCreacion = usuIdCreacion
					me.usuIdUltimaModificacion = usuIdUltimaModificacion
				Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			
			Public Function ObtenerTodo() As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerTodo]", New Object(,) {})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerUno(ByVal pmeId as Integer) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerUno]", New Object(,) {{"@pmeId", pmeId}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Function ObtenerxperIdCustom(ByVal perId as Integer) As DataTable
				ocdGestor = New Datos.Gestor
				Tabla = New DataTable

			    Try
					Tabla = ocdGestor.EjecutarReader("[PerfilMenu.ObtenerxperIdCustom]", New Object(,) {{"@perId", perId}})
				Catch ex As Exception
					Throw ex
				End Try
				
				Return Tabla
			End Function
			
			Public Sub Actualizar(ByVal pmeId as Integer, ByVal perId as Integer, ByVal menId as Integer, ByVal pmeActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal pmeFechaHoraCreacion as Date, ByVal pmeFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.Actualizar]", New Object(,) {{"@pmeId", pmeId}, {"@perId", perId}, {"@menId", menId}, {"@pmeActivo", pmeActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@pmeFechaHoraCreacion", pmeFechaHoraCreacion}, {"@pmeFechaHoraUltimaModificacion", pmeFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Eliminar(ByVal pmeId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.Eliminar]", New Object(,) {{"@pmeId", pmeId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub EliminarPadresSinHijos(ByVal perId as Integer)
				Try
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.EliminarPadresSinHijos]", New Object(,) {{"@perId", perId}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Insertar(ByVal perId as Integer, ByVal menId as Integer, ByVal pmeActivo as Boolean, ByVal usuIdCreacion as Integer, ByVal usuIdUltimaModificacion as Integer, ByVal pmeFechaHoraCreacion as Date, ByVal pmeFechaHoraUltimaModificacion as Date)
				Try
                    ocdGestor.EjecutarNonQuery("[PerfilMenu.Insertar]", New Object(,) {{"@perId", perId}, {"@menId", menId}, {"@pmeActivo", pmeActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@pmeFechaHoraCreacion", pmeFechaHoraCreacion}, {"@pmeFechaHoraUltimaModificacion", pmeFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
			End Sub
			
			Public Sub Actualizar()
                Try
                    If Me.pmeId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[PerfilMenu.Actualizar]", New Object(,) {{"@pmeId", pmeId}, {"@perId", perId}, {"@menId", menId}, {"@pmeActivo", pmeActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@pmeFechaHoraCreacion", pmeFechaHoraCreacion}, {"@pmeFechaHoraUltimaModificacion", pmeFechaHoraUltimaModificacion}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Sub Eliminar()
                Try
                    If Me.pmeId <> 0 Then
                        ocdGestor.EjecutarNonQuery("[PerfilMenu.Eliminar]", New Object(,) {{"@pmeId", pmeId}})
                    End If
                Catch ex As Exception
                    Throw ex
                End Try
            End Sub

			Public Function Insertar() as Integer
                Dim IdMax As Integer
                Try
                    IdMax = ocdGestor.EjecutarNonQueryRetornaId("[PerfilMenu.Insertar]", New Object(,) {{"@perId", perId}, {"@menId", menId}, {"@pmeActivo", pmeActivo}, {"@usuIdCreacion", usuIdCreacion}, {"@usuIdUltimaModificacion", usuIdUltimaModificacion}, {"@pmeFechaHoraCreacion", pmeFechaHoraCreacion}, {"@pmeFechaHoraUltimaModificacion", pmeFechaHoraUltimaModificacion}})
                Catch ex As Exception
                    Throw ex
                End Try
                Return IdMax
            End Function

		End Class
    End Namespace
End Namespace